lib/rewrite/rewrite-set-severity.c

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2019 Balabit
+ * Copyright (c) 2019 Kokan <kokaipeter@gmail.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+#include "rewrite-set-severity.h"
+#include "template/templates.h"
+#include "syslog-names.h"
+#include "scratch-buffers.h"
+
+#include <stdlib.h>
+
+typedef struct _LogRewriteSetSeverity LogRewriteSetSeverity;
+
+struct _LogRewriteSetSeverity
+{
+  LogRewrite super;
+  LogTemplate *severity;
+};
+
+static gint
+_convert_severity_as_number(GString *severity_text)
+{
+  char *endptr;
+  long int severity = strtol(severity_text->str, &endptr, 10);
+  if (0 == endptr)
```

Comment:
we have a scan_positive_int() function that is used by the date scanning routines and if I remember their reason for existance over strtol() was performance. I am not 100% sure that this is still valid, but I think we should use that function for this use-case, as this can be performance sensitive.

Reply:
**please disregard** this comment, scan_positive_int() is pretty awkward to use in this scenario, we also seem to have a parse-number module which is used by numeric template functions and which uses strtol().

Using parse_dec_number() is thus not a performance improvement, only a slight improvement in readability. I am working on adding set-facility() (someone asked for that recently) and I'll update this to use the same mechanism for parsing a number.

Reply:
:+1: